### DISCUSSION TITLE
DB persistence with seeders

### DISCUSSION THREAD

**Discussion by italianst4** [veelasky/laravel-hashid#21] [Q&A] [answered]
How do you generate persistent db hashes with database seeders?

**Comment by veelasky** [comment]
have you tried using 

```php
class User extends Model 
{
    use HashableId;

    protected $shouldHashPersist = true;
    
    // your hash column name
    protected $hashColumnName = 'hashid';
}
```

and make sure you are not using `Illuminate\Database\Console\Seeds\WithoutModelEvents` trait in your seeder class, since `laravel-hashid` utilize Model event.

**Comment by italianst4** [comment, marked as answer]
I realized the reason it wasn't working is because I was using direct DB inserts and not model generations.

The seeders were originally set up per Laravel docs (https://laravel.com/docs/9.x/seeding#writing-seeders).

I updated my seeders to seed with models.  That did the trick!

Thank you.
